src/sqlparser/src/ast/mod.rs

```diff
@@ -3240,48 +3240,61 @@ impl fmt::Display for CreateFunctionBody {
 pub struct CreateFunctionWithOptions {
     /// Always retry on network errors.
     pub always_retry_on_network_error: Option<bool>,
-}
-
-/// TODO(kwannoel): Generate from the struct definition instead.
-impl CreateFunctionWithOptions {
-    fn is_empty(&self) -> bool {
-        self.always_retry_on_network_error.is_none()
-    }
+    /// Use async functions (only available for JS UDF)
+    pub r#async: Option<bool>,
+    /// Call in batch mode (only available for JS UDF)
+    pub batch: Option<bool>,
 }
 
 /// TODO(kwannoel): Generate from the struct definition instead.
 impl TryFrom<Vec<SqlOption>> for CreateFunctionWithOptions {
     type Error = StrError;
 
     fn try_from(with_options: Vec<SqlOption>) -> Result<Self, Self::Error> {
-        let mut always_retry_on_network_error = None;
+        let mut options = Self::default();
         for option in with_options {
-            if option.name.to_string().to_lowercase() == "always_retry_on_network_error" {
-                always_retry_on_network_error = Some(matches!(
-                    option.value,
-                    SqlOptionValue::Value(Value::Boolean(true))
-                ));
-            } else {
-                return Err(StrError(format!("Unsupported option: {}", option.name)));
+            match option.name.to_string().to_lowercase().as_str() {
+                "always_retry_on_network_error" => {
+                    options.always_retry_on_network_error = Some(matches!(
+                        option.value,
+                        SqlOptionValue::Value(Value::Boolean(true))
+                    ));
+                }
+                "async" => {
+                    options.r#async = Some(matches!(
+                        option.value,
+                        SqlOptionValue::Value(Value::Boolean(true))
+                    ))
+                }
+                "batch" => {
```

Comment:
Shall we name it `batched` instead? 🤔

Reply:
We also use `batch=True` in external python udf